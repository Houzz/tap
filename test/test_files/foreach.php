<?php

$arr = array();
foreach ($arr as $k => $v) {
  __tap_check($k, "any");
  __tap_check($v, "any");
}
foreach ($arr as $v) {
  __tap_check($k, "any");
}

/**
 * @var vector<string>
 */
$vec = array("str1", "str2");
foreach ($vec as $k => $v) {
   __tap_check($k, "int");
   __tap_check($v, "string");
}
foreach ($vec as $v) {
   __tap_check($v, "string");
}

$map = array("str1" => 1, "str2" => 2);
foreach ($map as $k => $v) {
  __tap_check($k, "string");
  __tap_check($v, "int");
}
foreach ($map as $v) {
  __tap_check($v, "int");
}

// check undefined variable
foreach ($arr as $v) {
  $x = 1;
}
__tap_check($x, "int|null");

$a = array("str1"); // this is considered as tuple
foreach ($a as $k => $v) {
  __tap_check($k, "int");
  __tap_check($v, "any");
}
foreach ($a as $v) {
  __tap_check($v, "any");
}

