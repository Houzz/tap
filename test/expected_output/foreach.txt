37 E_NAME_UNDEFINED
